39891: optimize container filter when no special child folder types exist

labkey-kevink opened the pull request:
When no special container types exist in the set of containers, use a simple IN clause filter instead of joining to core.containers and filtering by type.

labkey-kevink committed:
Update ContainerFilter.java

labkey-jeckels commented on api/src/org/labkey/api/data/ContainerFilter.java:
```diff
@@ -214,13 +214,30 @@ protected SQLFragment _getSQLFragment(DbSchema schema, Container container, SQLF
+        // have children that match the types. If there are no children that match
+        // the set of included types, we can simplify the query to a simple IN clause
+        // instead of joining to core.Containers and filtering by container type.
+        Set<String> finalIncludedChildTypes = includedChildTypes;
```
I follow what's happening here and how Java can infer that it's effectively final, but perhaps better to use "final" keyword to make it clearer.

labkey-jeckels reviewed (approved):
One suggestion for clarity

labkey-kevink committed:
add final to finalIncludedChildTypes reference

We may clear the includedChildTypes below so we need to create a final reference to the original set of includedChildTypes
